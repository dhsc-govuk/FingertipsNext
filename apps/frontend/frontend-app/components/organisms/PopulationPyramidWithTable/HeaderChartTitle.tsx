import { H5 } from 'govuk-react';

interface HeaderChartTitleProps {
  title: string;
}

export const HeaderChartTitle = ({
  title,
}: Readonly<HeaderChartTitleProps>) => {
  return (
    <div style={{ margin: '0px', padding: '0px' }}>
      <H5 style={{ margin: '0px', padding: '0px' }}>{title}</H5>
    </div>
  );
};
