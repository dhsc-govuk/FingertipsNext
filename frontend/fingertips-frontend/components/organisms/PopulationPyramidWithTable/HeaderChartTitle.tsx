interface HeaderChartTitleProps {
  title: string;
}

export const HeaderChartTitle = ({
  title,
}: Readonly<HeaderChartTitleProps>) => {
  return (
    <div style={{ margin: '0px', padding: '0px' }}>
      <h3 style={{ margin: '0px', padding: '0px' }}>{title}</h3>
    </div>
  );
};
