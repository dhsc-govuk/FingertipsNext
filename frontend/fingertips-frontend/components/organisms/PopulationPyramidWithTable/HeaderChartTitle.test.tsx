import { render, screen } from '@testing-library/react';
import { HeaderChartTitle } from './HeaderChartTitle';
import '@testing-library/jest-dom';

describe('HeaderChartTitle Component', () => {
  test('renders correctly with a given title', () => {
    render(<HeaderChartTitle title="Test Chart Title" />);

    expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent(
      'Test Chart Title'
    );
  });

  test('renders correctly with an empty title', () => {
    render(<HeaderChartTitle title="" />);

    expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent('');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<HeaderChartTitle title="Snapshot Title" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
