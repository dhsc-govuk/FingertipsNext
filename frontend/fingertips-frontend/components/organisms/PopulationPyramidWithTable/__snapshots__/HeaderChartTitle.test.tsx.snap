// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderChartTitle Component matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-top: 0;
  margin-bottom: 15px;
}

@media print {
  .c0 {
    color: #000;
  }
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 20px;
  }
}

<div
    style="margin: 0px; padding: 0px;"
  >
    <h5
      class="c0"
      style="margin: 0px; padding: 0px;"
    >
      Snapshot Title
    </h5>
  </div>
</DocumentFragment>
`;
